### ISSUE TITLE
[MNT]: Deprecated / discourage less used Axes methods forwarding to Axis methods

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#23633]
### Summary

Discourage or deprecate

- `Axes.invert_xaxis()` / `Axes.invert_yaxis()`. Use `ax.xaxis.set_inverted(bool)` instead.
- `Axes.xaxis_inverted()` / `Axes.yaxis_inverted()`. Use `ax.xaxis.get_inverted()` instead.

These are simple and rare enough so that we do not need to lift the API from the Axis to the Axes.

I'm still not decided whether we should only discourage or deprecate. - It's the usual trade-off between backward-compatibility and making the API simpler / less redundant.

**Comment by story645** [comment]
I'm kinda surprised there isn't an `Axes.invert(which={x, y, both})` method since that seems like it'd be more consistent with the other methods on axes. Also what I'd propose replacing `invert_{xy}axis` with if you're going to deprecate those so there is still a method on axes and it is therefore discoverable in those docs.

**Comment by jklymak** [comment]
My problem with this is the `axis` API page is somewhat hard to discover.  Maybe the API top level page needs to mention this and a couple other modules as being key to understanding the library (we have `Axes` and `Figure` already mapped out pretty well)

**Comment by anntzer** [comment]
The problem of `Axes.invert` (or `Axes.invert_xaxis`) is that it can easily make code *less* robust, because if you factor out your block of code into a library function that takes an `ax` parameter, things will typically go wrong if the axis was already inverted to start with (see the changelog entry in https://github.com/matplotlib/matplotlib/pull/13330).  So I think we should just encourage the more robust pattern (`set_inverted(True)`).
I think invert_xaxis has been around for long enough that we should just discourage it without deprecating it.

**Comment by story645** [comment]
> set_inverted(True)

can this be added as a kwarg to `.set_tick_params`? Also then what about `Axes.inverted(inverted=False, which='both')` which mirrors [grid](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.grid.html?highlight=grid) and calls out to `set_inverted`

**Comment by anntzer** [comment]
It's certainly not a tick param.  I guess I don't really like `Axes.inverted(bool, which=...)`, but I probably don't feel as strongly about it.

**Comment by anntzer** [comment]
Not really, at least not any more than `plot([3, 4])` is a ticks operation in that it sets visible ticks at 3 and 4, I would say.

**Comment by timhoffm** [comment]
To make the discussion short, with my API authority I claim: We're not going to introduce any new Axes-level API for this. It's either the existing methods or going down to Axis via `ax.xaxis` / `ax.yaxis`.

Sorry for being terse, I can motivate this in detail if there is interest, but I don't have the time right now.

**Comment by story645** [comment]
I think consensus is discourage not deprecate - Jody and me for discoverability reasons, Anthony 'cause it's too widely used?

**Comment by timhoffm** [comment]
Closed by #29074.
